src/xform_to_datalog/authorization_logic_datalog_facts.h

```diff
@@ -0,0 +1,45 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#ifndef SRC_XFORM_TO_DATALOG_AUTHORIZATION_LOGIC_DATALOG_FACTS_H_
+#define SRC_XFORM_TO_DATALOG_AUTHORIZATION_LOGIC_DATALOG_FACTS_H_
+
+#include <optional>
+#include <string>
+
+namespace raksha::xform_to_datalog {
+
+class AuthorizationLogicDatalogFacts {
+ public:
+  // Creates the datalog logic facts from the given authorization logic program.
+  // Returns std::nullopt if there is any error due to processing of the program.
+  //
+  static std::optional<AuthorizationLogicDatalogFacts> create(
+      const std::string& path,							      
+      const std::string& authorization_logic_filename);
```

Comment:
I think both of these strings are expected to be read in their entirety, by the `create` method in the common case. As such, it's probably better to just pass a `std::string` to remove lifetime concerns and make these eligable for copy elision.

Reply:
We don't make a copy of these strings internally. If that is the case, isn't `const &` preferred?

Even in the current code, we pass in some strings that need not be copied.

Reply:
You are right,this is not a good case for copy elision, sorry for the red herring. In this case, however, the standard recommendation is to use `absl::string_view` rather than `const std::string &` for function parameters.